src/test/java/tests/FilterParserTests.java

```diff
@@ -72,6 +72,10 @@ public void unexpectedEOFs() {
             Parser.parse("a(b) ||");
             fail("Inputs which end unexpectedly should throw a parse exception");
         } catch (ParseException ignored) {}
+        try {
+            Parser.parse("a:b; c:d;e");
+            fail("Inputs which end unexpectedly should throw a parse exception");
```

Comment:
Hmm, didn't know why I put it there in the first place. Thanks for the correction.